### PR TITLE
Add support for labels during build with compose

### PR DESCRIPTION
**Pull request by ColinHebert**
As per requested in https://github.com/docker/compose/pull/4735, this is an update of the docker-compose schema to allow setting labels during the build process.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Oh! CI is failing because you need to run `go generate` for this. However, I think this change may have to go in a 3.3 format, not in 3.2 (not exactly sure when versions should be bumped for such changes);

````
21:42:22 The result of `go generate github.com/docker/docker/cli/compose/schema` differs
21:42:22 
21:42:22  M cli/compose/schema/bindata.go
21:42:22 
21:42:22 Please run `go generate github.com/docker/docker/cli/compose/schema`
````

**Comment by ColinHebert** [comment]
@shin-, @dnephin, what is you recommendation? Should a new schema be created or can that addition be made to 3.2?

File: cli/compose/schema/bindata.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (fi bindataFileInfo) Sys() interface{} {
...
...a4\x00\x07\xef\xee\x75\xb4\xd2\x48\x28\xd7\x57\x97\xf3\x7b\xcd\xba\xa3\x9a\x0f\xf9\x19\x41\xfd\x81\x2e\xfe\x26\x35\x80\xf4\x85\xac\xa3\xf3\xa0\x0a\x2a\x81\xde\x04\x5c\xe0\xcd\x77\xcc\xc4\x9e\x6c\x81\xaa\x36\x6b\x80\xd8\x51\x99\xfb\xe2\xe2\xb7\x8d\xf8\x90\x30\x8b\x03\x12\x95\xa4\x5a\xea\x74\xcc\x1e\xa9\xa6\xde\x1a\x9c\x4c\x8f\x22\x92\xf0\x38\x22\xa6\x75\x5c\xf7\x8e\x42\xd5\x1b\x0e\x93\x1d\x95\xe5\x4f\xdf\x7b\xde\xf9\xd7\x94\x53\xf8\x52\x72\x1f\xe8\xf5\x6f\x43\x02\x51\x7d\x1e\x7a\xe6\xf5\xe0\xab\x6c\x76\x88\x83\xaf\x22\x96\xd3\xbf\x69\xdf\xdd\x11\x81\xaf\xcf\xbf\xb2\x13\xbc\x03\x5c\xba\x4f\x3c\x22\xd8\xd2\x51\xfd\x03\x2d\x7f\x93\x44\xfc\x79\xf9\xe5\x4c\xb3\xac\x3c\xbb\xbc\x68\x4e\xa5\xc4\xec\x31\x7e\xc7\x91\x8d\xd5\x70\xc9\x3c\x5f\xda\x8d\x61\x79\x6a\x7a\xd1\x93\x04\xc6\xba\xce\xa6\x9d\x13\xa7\x2d\x5f\x30\xc3\x1f\xdf\x4d\x14\x9f\xa9\xd7\x6d\x3f\x08\xb5\x17\x98\x0c\xf9\x63\xea\x74\xac\xbd\x77\x2f\x3f\x17\x0b\x80\x9a\xc5\x7f\xf1\xf1\x98\xb1\x93\x1f\x2f\x06\x21\xdf\xc7\xd3\xbd\xf6\xc3\xaf\x6c\xe4\x1f\x87\xa4\x7d\x79\x6d\x41\x4a\x66\x37\xf1\xa1\x30\x7a\x3f\x29\x73\x67\x8b\xfd\xa7\x5d\x99\x1f\xae\x56\xf6\xdf\xe6\x33\xbc\xd5\x69\xf5\x57\x00\x00\x00\xff\xff\x78\x30\xec\x51\x0e\x2b\x00\x00")
```
Comment (thaJeztah) [inline cli/compose/schema/bindata.go]: Interesting; wondering why the old versions also god updated

Comment (ColinHebert) [inline cli/compose/schema/bindata.go]: Not really sure, apparently it's still failing :/

Steps I ran:
```
$ docker run --rm -ti -v $(pwd):/go/src/github.com/docker/docker golang /bin/bash
# go get -u github.com/jteeuwen/go-bindata/...
# go generate github.com/docker/docker/cli/compose/schema
```

Comment (ColinHebert) [inline cli/compose/schema/bindata.go]: I tried to get the binary code and unzip it (`pbpaste | tr '\\x' ' ' | xxd -r -p | gunzip`), and the extracted content is strictly identical.

I'm suspecting some slight differences in gzip libs :/

**Review by dnephin — APPROVED** [review]
LGTM

We can create the new schema before the release.

**Comment by ColinHebert** [comment]
Awesome, @dnephin @thaJeztah is there anything else I need to do?

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Comment by thaJeztah** [comment]
@dnephin looks like these changes had to be moved to the 3.3 schema?

**Comment by dnephin** [comment]
They will need to be, yes. This is in 17.06 is that right? I've added the milestone

**Comment by thaJeztah** [comment]
Discussing with @dnephin on slack; we can make the schema change after https://github.com/moby/moby/pull/32743 is merged

**Pull request merged**